linux_sql/scripts/psql_docker.sh

```diff
@@ -0,0 +1,60 @@
+#! /bin/sh
+
+# Capture CLI arguments
+cmd=$1
+db_username=$2
+db_password=$3
+
+# Start docker
+# Make sure you understand the double pipe operator
+sudo systemctl status docker || systemctl start docker
+
+# Check container status (try the following cmds on terminal)
+docker container inspect jrvs-psql
+container_status=$?
+
+# User switch case to handle create|stop|start opetions
+case $cmd in
+  create)
+
+  # Check if the container is already created
+  if [ $container_status -eq 0 ]; then
+		echo 'Container already exists'
+		exit 1
+	fi
+
+  # Check # of CLI arguments
+  if [ $# -ne 3 ]; then
+    echo 'Create requires username and password'
+    exit 1
+  fi
+
+  # Create container
+	docker volume pgdata
+  # Start the container with the given username and password
+	docker run --name jrvs-psql -e POSTGRES_PASSWORD=$db_password -d -v pgdata:/var/lib/postgresql/data -p 5432:5432 -e POSTGRES_USER=$db_username postgres:9.6-alpine
+  # Make sure you understand what's `$?`
+  # check the exit status of the last command
+	exit $?
+	;;
+
+  start|stop)
+  # Check instance status; exit 1 if container has not been created
+  if [ $container_status -ne 0 ]; then
+    echo 'Container does not exist'
+    exit 1
+  fi
+
+
+  # Start or stop the container
+	docker container $cmd jrvs-psql
+	exit $?
+	;;
```

Comment:
these should be within an elif in the above conditional block, otherwise they will try to execute regardless of the container existing or not.